presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/TestMergeAdjacentWindows.java

```diff
@@ -0,0 +1,123 @@
+                .on(p -> p.values(p.symbol("a", BIGINT)))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testDoesNotFireOnPlanWithSingleWindowNode()
+            throws Exception
+    {
+        tester.assertThat(new MergeAdjacentWindows())
+                .on(p -> p.window(
+                        new WindowNode.Specification(
+                                ImmutableList.of(p.symbol("a", BIGINT)),
+                                ImmutableList.of(),
+                                ImmutableMap.of()),
+                        ImmutableMap.of(p.symbol("avg_1", BIGINT),
+                                new WindowNode.Function(new FunctionCall(QualifiedName.of("avg"), ImmutableList.of()), signature, frame)),
+                        p.values(p.symbol("a", BIGINT))))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testDoesNotFireForDistinctAdjacentWindowSpecifications()
+    {
+        tester.assertThat(new MergeAdjacentWindows())
+                .on(p ->
+                        p.window(
+                                new WindowNode.Specification(ImmutableList.of(p.symbol("a", BIGINT)), ImmutableList.of(), ImmutableMap.of()),
+                                ImmutableMap.of(p.symbol("avg_1", BIGINT), new WindowNode.Function(new FunctionCall(QualifiedName.of("avg"), ImmutableList.of()), signature, frame)),
+                                p.window(
+                                        new WindowNode.Specification(ImmutableList.of(p.symbol("b", BIGINT)), ImmutableList.of(), ImmutableMap.of()),
+                                        ImmutableMap.of(p.symbol("sum_1", BIGINT),
+                                                new WindowNode.Function(new FunctionCall(QualifiedName.of("sum"), ImmutableList.of()), signature, frame)),
+                                        p.values(p.symbol("a", BIGINT), p.symbol("b", BIGINT))
+                                )
+                        ))
+                .doesNotFire();
+    }
+
+    @Test
+    public void testIdenticalAdjacentWindowSpecifications()
+            throws Exception
+    {
+        tester.assertThat(new MergeAdjacentWindows())
+                .on(p ->
+                        p.window(
+                                new WindowNode.Specification(ImmutableList.of(p.symbol("a", BIGINT)), ImmutableList.of(), ImmutableMap.of()),
+                                ImmutableMap.of(p.symbol("avg_1", BIGINT), new WindowNode.Function(new FunctionCall(QualifiedName.of("avg"), ImmutableList.of()), signature, frame)),
+                                p.window(
+                                        new WindowNode.Specification(ImmutableList.of(p.symbol("a", BIGINT)), ImmutableList.of(), ImmutableMap.of()),
+                                        ImmutableMap.of(p.symbol("sum_1", BIGINT),
+                                                new WindowNode.Function(new FunctionCall(QualifiedName.of("sum"), ImmutableList.of()), signature, frame)),
```

Comment:
Add references to arguments. Have a look here: https://github.com/prestodb/presto/pull/7300/files#diff-395829819374698c764b6244098a5233R112 and corresponding https://github.com/prestodb/presto/pull/7300/files#diff-395829819374698c764b6244098a5233R115.